security/advancedtls/advancedtls.go

```diff
@@ -611,6 +618,7 @@ func buildVerifyFunc(c *advancedTLSCreds,
 				return err
 			}
 			leafCert = rawCertList[0]
+			*peerVerifiedChains = chains
```

Comment:
We would not want to do this until right before the `return nil` on line 643, otherwise we might set these as verified chains then return an error if we find they are revoked by a CRL. A custom verification func that was supplied could fail as well.

Reply:
Thanks for the review. addressed it in the next patch